### DISCUSSION TITLE
Power consumption sensor

### DISCUSSION THREAD

**Discussion by GeraldPape** [flobz/psa_car_controller#664] [Ideas]
first, thanks for your work on this tool. 

I use home assistant to visualize the cars status. especially the battery. In my energy portal I would like to see how much kWh I charged the car. 
Is there a possibility to provide a sensor to home assistant that gets the current consumption while loading the car? So I can use the sensor in the energy dashboard and see how much loading my car costs ;-)

Thank you!

**Comment by sod75** [comment]
Would indeed be nice. 
I'm no expert but I don't think cars provide that info unfortunately judging by this : [Data Catalog](https://developer.groupe-psa.io/connected-vehicles/data-catalog/#article)
I guess the solution is getting this from the charge point...

**Reply by sod75** [reply]
But it does store it in the database....
Here's a discussion to get the data :[356]( https://github.com/flobz/psa_car_controller/issues/356)

**Reply by sod75** [reply]
I had a look in the code as I wondered where we got the consumption from if the car doesn't provide it.

It's a calculation based on averages from mileage, time charging, etc so an approximation but not that actual values

in psa_car_controller/web/assets/clientside.js:

```
function updateCardsValue (data) {
  const res = {}
  let avgPriceKw
  let avgC02 = new Avg(); let avgKw = new Avg(); const avgTime = new Avg()
  const avgPrice = new Avg()
  data.chargings.forEach(function (charge) {
    const diff = ((new Date(charge.stop_at)) - (new Date(charge.start_at))) / 3600000
    avgKw.addValue(charge.kw)
    avgC02.addValue(charge.co2)
    avgPrice.addValue(charge.price)
    if (diff > 0) {
      avgTime.addValue(diff)
    }
  })
  if (data.chargings.length > 0) {
    avgKw = avgKw.average()
    avgC02 = avgC02.average()
    avgPriceKw = avgPrice.average() / avgKw
    res.avg_emission_kw = avgC02
    res.avg_chg_speed = avgKw / avgTime.average()
  }
  if (data.trips.length > 0) {
    const totalDistance = Math.abs(data.trips[data.trips.length - 1].mileage - data.trips[0].mileage)
    res.avg_consum_kw = Avg.getAverageFromKey(data.trips, 'consumption_km')
    res.elec_consum_kw = totalDistance * res.avg_consum_kw / 100
  }
  if (data.trips.length > 0 && data.chargings.length > 0) {
    res.avg_emission_km = res.avg_emission_kw * res.avg_consum_kw / 100
    res.elec_consum_price = avgPriceKw * res.elec_consum_kw
    res.avg_consum_price = avgPriceKw * res.avg_consum_kw
  }
  for (const [key, value] of Object.entries(res)) {
    document.getElementById(key).innerHTML = nbFormat(value)
```
